pkg/certmanager/certmanager.go

```diff
@@ -184,7 +184,11 @@ func (m *manager) waitForCertificateRequest(ctx context.Context, log logr.Logger
 
 		log.V(3).Info("waiting for CertificateRequest to become ready")
 
-		w := <-watcher.ResultChan()
+		var w watch.Event
+		for w.Object == nil {
```

Comment:
Suggestion: instead of doing this couldn't you just do
```
cr, ok  := w.Object.(*cmapi.CertificateRequest)
if !ok {
  continue
}
```

That would I guess catch the nil edge case as well as any other where the object wouldn't be a CR

Reply:
Thanks, this is a much better way of doing it :+1: